test/system/inspect.spec.ts

```diff
@@ -100,4 +105,61 @@ describe('inspect', () => {
       expect(result.plugin.targetFile).toEqual(expectedTargetFile);
     });
   });
+
+  describe('error scenarios', () => {
+    const mockedExecuteSync = jest.spyOn(subProcess, 'executeSync');
+    const mockedExecute = jest.spyOn(subProcess, 'execute');
+
+    afterEach(() => {
+      mockedExecuteSync.mockClear();
+      mockedExecute.mockClear();
+    });
+
+    describe('manifest file is empty', () => {
+      it('should throw EmptyManifestError', async () => {
+        mockedExecute.mockResolvedValueOnce('Python 3.9.5');
+        mockedExecute.mockRejectedValueOnce(
+          'No dependencies detected in manifest.'
+        );
+        const manifestFilePath = 'path/to/requirements.txt';
+
+        try {
+          await inspect('.', manifestFilePath);
+          throw new Error('expected to throw EmptyManifestError but did not');
```

Comment:
this clunky logic was put in favour of [toThrow](https://jestjs.io/docs/expect#tothrowerror) because this didn't work
```javascript
expect(async () => {
  await inspect('.', manifestFilePath);
}).toThrowError(EmptyManifestError);

not sure why it wasn't working, there should be just a promise to await but it acted as there were some callbacks that thre that were not awaited :/

Reply:
found better way to assert in https://github.com/snyk/snyk-python-plugin/pull/158/commits/b1c90d913a8fbe1b9bc1518b515d3b4678bdb3d2